chapter10/ajd/exercise2.pl

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Benchmark qw( :all );
+
+sub naive {
+  chdir;
+  my @sorted = sort { -s $a <=> -s $b } glob '*';
+  # Explicit return value to stop the array being returned
+  # Turns out Benchmark goes off and tests all the scripts that this returns
+  return 1;
+}
+
+sub schwartz {
+  chdir;
+  my @schwartz = map { $_->{name} }
+                 sort { $a->{size} <=> $b->{size} }
+                 map { { name => $_, size => -s $_ } }
+                 glob '*';
+  return 1;
+}
+
+# Run for 60 seconds and compare the results
+my $result = cmpthese( -60 , {
+    'naive' => &naive,
+    'schwartz' => &schwartz,
```

Comment:
Is this not why you were seeing the weird benchmark behaviour? It looks like you're assigning these with the results of the subroutine call rather than assigning them the subref (`&naive` vs `\&naive`).